src/main/java/com/minecolonies/coremod/client/gui/ViewFilterableList.java

```diff
@@ -144,23 +146,23 @@ private void switchClicked(@NotNull final Button button)
             button.setText(OFF);
             if (isInverted)
             {
-                building.addItem(id, allItems.get(row));
+                building.getModuleViewMatching(ItemListModuleView.class, view -> view.getId().equals(id)).ifPresent(m -> m.addItem(allItems.get(row)));
             }
             else
             {
-                building.removeItem(id, allItems.get(row));
+                building.getModuleViewMatching(ItemListModuleView.class, view -> view.getId().equals(id)).ifPresent(m -> m.removeItem(allItems.get(row)));
             }
         }
         else
         {
             button.setText(ON);
             if (isInverted)
             {
-                building.removeItem(id, allItems.get(row));
+                building.getModuleViewMatching(ItemListModuleView.class, view -> view.getId().equals(id)).ifPresent(m -> m.removeItem(allItems.get(row)));
             }
             else
             {
-                building.addItem(id, allItems.get(row));
+                building.getModuleViewMatching(ItemListModuleView.class, view -> view.getId().equals(id)).ifPresent(m -> m.addItem(allItems.get(row)));
```

Comment:
This looks like an awefull lot of repetition.
You can extract a method here:

``` java
private void handleItemListInteraction(final ResourceLocation listId, final int row, final BiConsumer<ItemListModuleView, ItemStorage> action) {
    building.getModuleViewMatching(ItemListModuleView.class, view -> view.getId().equals(listId)).ifPresent(m -> action.accept(m, allItems.get(row)));
}
```

Which would turn this section into: 
``` java
   BiConsumer<ItemListModuleView, ItemStorage> action = ItemListModuleView::addItem;
   if ((button.getTextAsString().equals(ON) && !isInverted) || (button.getTextAsString().equals(OFF) && isInverted))
   {
       action = ItemListModuleView::removeItem;
   }

   button.setText((button.getTextAsString().equals(ON) ? OFF : ON);
   handleItemListInteraction(id, row, action);
```

Which is in my eyes much cleaner and shorter.

Reply:
I will rewrite the entire Window side of this (including this class) in the next PR. I'd like to avoid changing this now just to scrap all of this later again.

Reply:
Okey